docs/_guides/nifi-jms-jndi.md

```diff
@@ -0,0 +1,695 @@
+* Step 3 - Developing a Solace JMS application to do message PubSub via JNDI
+* Step 4 - Obtain Apache NiFi
+* Step 5 - Developing a new NiFi JMS ControllerService to support JNDI
+* Step 6 - Deploying NiFi JMS ControllerService
+
+### Description of Resources Required
+
+This integration guide will demonstrate creation of Solace JMS custom sender and receiver and configuring the sender and receiver to send and receive messages. This section outlines the resources that are required, created and used in the subsequent sections.
+
+#### Solace Resources
+
+The following Solace Message Router resources are required.
+
+
+
+|Resource|Value|Description|
+|---|---|---|
+|Solace Message Router IP:Port|__IP:Port__|The IP address and port of the Solace Message Router message backbone. This is the address client’s use when connecting to the Solace Message Router to send and receive message. This document uses a value of __IP:PORT__.|
+|Message VPN|default|A Message VPN, or virtual message broker, to scope the integration on the Solace message router.|
+|Client Username|nifi|The client username.|
+|Client Password||Optional client password. |
+|Solace Queue|toNifi|Solace destination of persistent messages consumed by NiFi|
+|Solace Queue|fromNifi|Solace destination of persistent messages published from NiFi|
+|JNDI Connection Factory|/jms/cf/default|The JNDI Connection factory for controlling Solace JMS connection properties|
+|JNDI Queue Name|/JNDI/Q/toNifi|The JNDI name of the queue used in the samples|
+|JNDI Queue Name|/JNDI/Q/fromNifi|The JNDI name of the queue used in the samples|
+
+###	Step 1 – Obtain access to Solace message router and JMS API
+
+The Solace messaging router can be obtained one of 2 ways.     
+1.	If you are in an organization that is an existing Solace customer, it is likely your organization already has Solace Message Routers and corporate policies about their use.  You will have to contact your middleware operational team in regards to access to a Solace Message Router.
+2.	If you are new to Solace or your company does not have development message routers, you can obtain a trail Solace Virtual Message Router (VMR) from the [Solace Developer Portal Downloads]({{ site.links-downloads }}){:target="_top"}. For help getting started with your Solace VMR you can refer to [Solace VMR Getting Started Guides]({{ site.links-vmr-getstarted }}){:target="_top"}.
+
+The Solace JMS jars are required.  They can be obtained on [Solace Developer Portal Downloads]({{ site.links-downloads }}){:target="_top"} or from [Maven Central]({{ site.links-jms-maven }}){:target="_blank"}.
+
+##### Solace JMS Dependencies
+
+The easiest way to integrate Solace and NiFi is using the client libraries available via public [Maven Repositories](https://mvnrepository.com/artifact/com.solacesystems){:target="_blank"}. You can downloaded the libraries into a directory and reference them directly, the following resources are all required:
```

Comment:
Better to use this link: [Maven Central]({{ site.links-jms-maven }}){:target="_blank"}

You did that two lines above as an example.